src/rpc/core_rpc_server.cpp

```diff
@@ -478,18 +478,25 @@ namespace cryptonote
       bool r = m_core.get_pool_transactions(pool_txs);
       if(r)
       {
-        for (std::list<transaction>::const_iterator i = pool_txs.begin(); i != pool_txs.end(); ++i)
+        // sort to match original request
+        std::list<transaction> sorted_txs;
+        std::list<cryptonote::transaction>::const_iterator i;
+        for (const crypto::hash &h: vh)
         {
-          crypto::hash tx_hash = get_transaction_hash(*i);
-          std::list<crypto::hash>::iterator mi = std::find(missed_txs.begin(), missed_txs.end(), tx_hash);
-          if (mi != missed_txs.end())
+          if (std::find(missed_txs.begin(), missed_txs.end(), h) == missed_txs.end())
           {
-            pool_tx_hashes.insert(tx_hash);
-            missed_txs.erase(mi);
-            txs.push_back(*i);
+            // core returns the ones it finds in the right order
```

Comment:
Perhaps we should insert a check like this?
```cpp
if (get_transaction_hash(txs.front()) != h)
{
  res.status = "Failed: tx hash mismatch";
  return true;
}
```

Reply:
OK, done

Reply:
Note that the wallet now handles out of order txes anyway.